docs/LANGUAGE.md

```diff
@@ -0,0 +1,767 @@
+
+Opalog queries are assertions on data stored in OPA. These queries can be used to define policies that enumerate instances of data that violate the expected state of the system.
+
+## Why use Opalog?
+
+Use Opalog for defining policy that is easy to read and write.
+
+Opalog focuses on providing powerful support for referencing nested documents and ensuring that queries are correct and unambiguous.
+
+Opalog is declarative so policy authors can focus on what queries should return rather than how queries should be executed. These queries are simpler and more concise than the equivalent in an imperative language. Like other applications which support declarative query languages, OPA is able to optimize queries to improve performance, e.g., indexing, concurrent evaluation, reordering, etc.
+
+## The Basics
+
+This section introduces the main aspsects of Opalog.
+
+The simplest rule contains a single expression and is defined in terms of a [Scalar Value](#scalar-values):
+
+```opalog
+pi :- 3.14159
+```
+
+Rules define the content of documents. We can query for the content of the "pi" document generated by the rule above:
+
+```
+pi
+# 3.14159
+```
+
+Rules can also be defined in terms of [Composite Values](#composite-values):
+
+```opalog
+rect :- {"width": 2, "height": 4}
+```
+
+The result:
+
+```
+rect
+# {"width": 2, "height": 4}
+```
+
+Many expressions are defined in terms of [Equality](#equality). These expressions can be thought of as assertions. The simplest type of assertion takes two scalar values:
+
+```opalog
+v :- 42 = "the meaning of life"
+```
+
+If we query for the contents of "v" we see the expression has been evaluated:
+
+```
+v
+# false
+```
+
+The order of operands in an equality expression does not matter:
+
+```opalog
+u :- "the meaning of life" = 42
+```
+
+The result is the same:
+
+```
+u
+# false
+```
+
+We can define rules in terms of [Variables](#variables) as well:
+
+```opalog
+t :- x = 42, y = 41, x > y
+```
+
+Multiple expressions are separated by the comma (",") character. In order for the rule to be true, all of the expressions in the rule must true for some set of variable bindings. There may be multiple sets of variable bindings that make the rule true. The body of a rule can be understood intuitively as "\<expression-1> AND \<expression-2> AND ... AND \<expression-N>".
+
+When we query for the contents of "t" we see the obvious result:
+
+```
+t
+# true
+```
+
+The order of expressions in a rule does not affect the document's content:
+
+```opalog
+s :- x > y, y = 41, x = 42
+```
+
+The query result is the same:
+
+```
+s
+# true
+```
+
+Opalog supports [References](#references) to nested documents. For example:
+
+```opalog
+sites :- [{"name": "prod"}, {"name": "smoke1"}, {"name": "dev"}]
+r :- sites[].name = "prod"
+```
```

Comment:
I think we want this to be sites[i].name = "prod".  At least at this point, the empty braces are confusing.  And here we probably want to say that we are implicitly enumerating the elements of the sites array.